src/experimental/HomeChainCoordinator.sol

```diff
@@ -0,0 +1,363 @@
+
+        // TODO: This needs to come from the metadata itself as this will keep on changing
+        bytes32 networkPublicKey;
+        // 6. Store PSBT metadata
+        PSBTMetadata memory psbtMetaData = PSBTMetadata({
+            isMinted: true,
+            chainId: metadata.chainId,
+            user: metadata.receiverAddress,
+            lockedAmount: metadata.lockedAmount,
+            nativeTokenAmount: metadata.nativeTokenAmount,
+            networkPublicKey: networkPublicKey,
+            psbtData: _psbtData
+        });
+        btcTxnHash_psbtMetadata[_btcTxnHash] = psbtMetaData;
+
+        // 7. Send message through LayerZerobytes memory payload
+        bytes memory payload = abi.encode(
+            metadata.chainId, metadata.receiverAddress, _btcTxnHash, metadata.lockedAmount, metadata.nativeTokenAmount
+        );
+
+        // TODO: Create a function to get the correct MessageFee for the user
+        _lzSend(
+            _dstEid,
+            payload,
+            _options,
+            // Fee in native gas and ZRO token.
+            MessagingFee(msg.value, 0),
+            // Refund address in case of failed source message.
+            _refundAddress
+        );
+
+        emit MessageSent(_dstEid, _btcTxnHash, msg.sender, block.timestamp);
+    }
+
+    /**
+     * @dev Validates PSBT data and extracts metadata.
+     * @param _psbtData The PSBT data to validate.
+     * @return metadata The extracted transaction metadata.
+     */
+    function _validatePSBTData(bytes calldata _psbtData)
+        internal
+        view
+        returns (BitcoinTxnParser.TransactionMetadata memory)
+    {
+        if (_psbtData.length == 0) {
+            revert InvalidPSBTData();
+        }
+
+        // Parse transaction outputs and metadata
+        bytes memory opReturnData = BitcoinTxnParser.decodeBitcoinTxn(_psbtData);
+        BitcoinTxnParser.TransactionMetadata memory metadata = BitcoinTxnParser.decodeMetadata(opReturnData);
+
+        // Validate amounts
+        if (metadata.nativeTokenAmount > maxGasTokenAmount) {
+            revert InvalidAmount(metadata.nativeTokenAmount);
+        }
+        if (metadata.lockedAmount < minBTCAmount) {
+            revert InvalidAmount(metadata.lockedAmount);
+        }
+
+        // Validate receiver address
+        if (metadata.receiverAddress == address(0)) {
+            revert InvalidDestination(metadata.receiverAddress);
+        }
+
+        return metadata;
+    }
+
+    /**
+     * @dev Sends a message for a specific receiver with the given BTC transaction hash.
+     * @param _receiver The address of the receiver.
+     * @param _btcTxnHash The BTC transaction hash.
+     * @param _options LayerZero message options.
+     * @param _refundAddress The address to refund in case of failure.
+     */
+    function sendMessageFor(address _receiver, bytes32 _btcTxnHash, bytes calldata _options, address _refundAddress)
```

Comment:
remind me why we have this function?

Reply:
This is for the case when the relayed message might get dropped and doesn't reach the BaseChainCoordinator. In such scenario the user will be able to request the message to be sent again. Since the _btcTxnHash should already exist on HomeChainCoordinator, the check `isMinted` is removed as it would be set to true in the previous request.